buildkite/bazelci.py

```diff
@@ -1181,10 +1186,16 @@ def fetch_bazelcipy_command():
     return "curl -s {0} -o bazelci.py".format(bazelcipy_url())
 
 
-def upload_project_pipeline_step(project_name, git_repository, http_config, file_config):
+def upload_project_pipeline_step(project_name, git_repository, http_config, file_config, test_incompatible_flags):
     pipeline_command = ("{0} bazelci.py project_pipeline --project_name=\"{1}\" " +
-                        "--use_but --git_repository={2}").format(python_binary(), project_name,
+                        "--git_repository={2}").format(python_binary(), project_name,
                                                                  git_repository)
+    if  test_incompatible_flags:
+        incompatible_flags = os.environ.get("INCOMPATIBLE_FLAGS", "").split()
+        for flag in incompatible_flags:
+            pipeline_command += " --incompatible_flag=" + flag
+    else:
+        pipeline_command += " --use_but"
```

Comment:
There is no case in which we want to use the custom bazel binary and the flags?

Reply:
Yes, in the current plan, we don't need to test custom bazel binary with incompatible flags.